src/base/stack.c

```diff
@@ -1061,7 +1061,11 @@ function_stack_entry *xdebug_add_stack_frame(zend_execute_data *zdata, zend_op_a
 	}
 	zdata = EG(current_execute_data);
 
-	tmp = xdmalloc (sizeof (function_stack_entry));
+	if (XG_BASE(stackPool)) {
+		tmp = XG_BASE(stackPool) + level;
```

Comment:
This is potentially scary, as you don't check whether the stack pool is big enough. The level can be higher as the max_nesting_level setting may change. I also don't see where you check this where the function stack entries get freed. Potentially Xdebug now ends up freeing function_stack_entries that were allocated in the pool.